sprocs/calculate_assessment_question_stats.psql

```diff
@@ -0,0 +1,190 @@
+        WHERE e.role = 'Student'
+        GROUP BY
+            ai.user_id,
+            ai.assessment_id
+    ),
+    mean_assessment_scores AS (
+        SELECT
+            mean_assessment_scores_without_quintiles.user_id,
+            mean_assessment_scores_without_quintiles.assessment_id,
+            mean_assessment_scores_without_quintiles.user_score_perc,
+            ntile(5) OVER (
+              PARTITION BY mean_assessment_scores_without_quintiles.assessment_id
+              ORDER BY mean_assessment_scores_without_quintiles.user_score_perc
+            ) as quintile
+        FROM
+            mean_assessment_scores_without_quintiles
+    ),
+    assessment_question_users AS (
+        SELECT
+            aq.question_id,
+            ai.user_id,
+            aq.assessment_id,
+            aq.id AS assessment_question_id
+        FROM
+            instance_questions AS iq
+            JOIN assessment_instances AS ai ON (iq.assessment_instance_id=ai.id)
+            JOIN assessment_questions AS aq ON (iq.assessment_question_id=aq.id)
+            JOIN assessments AS a ON (ai.assessment_id = a.id)
+            JOIN course_instances AS ci ON (a.course_instance_id = ci.id)
+            JOIN enrollments AS e ON (ai.user_id = e.user_id AND ci.id = e.course_instance_id)
+        WHERE
+            aq.deleted_at IS NULL
+            AND aq.id = assessment_question_id_var
+            AND e.role = 'Student'
+        GROUP BY
+            ai.user_id,
+            aq.question_id,
+            aq.id,
+            aq.assessment_id
+    ),
+    question_score_quintiles AS (
+        SELECT
+            assessment_question_users.assessment_question_id,
+            mean_assessment_scores.quintile AS user_quintile,
+            avg(mean_question_scores.user_score_perc) AS score_perc
+        FROM
+            assessment_question_users
+            JOIN mean_assessment_scores ON (
+                mean_assessment_scores.user_id = assessment_question_users.user_id
+                AND mean_assessment_scores.assessment_id = assessment_question_users.assessment_id
+            )
+            JOIN mean_question_scores ON (
+                mean_question_scores.assessment_question_id = assessment_question_users.assessment_question_id
+                AND mean_question_scores.user_id = assessment_question_users.user_id
+            )
+        GROUP BY
+            mean_assessment_scores.quintile,
+            assessment_question_users.assessment_question_id
+        ORDER BY
+            mean_assessment_scores.quintile
+    ),
+    question_score_quintiles_condensed AS (
+        SELECT
+            question_score_quintiles.assessment_question_id,
+            array_agg(question_score_quintiles.score_perc) AS quintile_scores
+        FROM question_score_quintiles
+        GROUP BY
+            question_score_quintiles.assessment_question_id
+    ),
+    aq_stats_orig AS (
```

Comment:
This is our original set of statistics where we average over users and then do a secondary average, rather than a simple average over all instance questions.